File: src/main/java/omerozturk/hrms/business/abstracts/SystemUserService.java
package omerozturk.hrms.business.abstracts;

import omerozturk.hrms.core.utilities.result.DataResult;
import omerozturk.hrms.core.utilities.result.Result;
import omerozturk.hrms.entities.concretes.SystemUser;

import java.util.List;
import java.util.Optional;

public interface SystemUserService {
    DataResult<List<SystemUser>> getAll();
    Result add(SystemUser systemUser);
    Result delete(SystemUser systemUser);
    DataResult<SystemUser> getById(int systemUserId);
}

File: src/main/java/omerozturk/hrms/uiPresentation/src/services/systemServices/systemUserService.js
import axios from "axios";

export default class SystemUserService {
    getAll() {
        return axios.get("http://localhost:8080/api/systemusers/getAll" );
    }
    getById(id) {
        return axios.get("http://localhost:8080/api/systemusers/getbyid?systemUserId=" + id);
    }

    add(systemUser) {
        return axios.post('http://localhost:8080/api/systemusers/add', systemUser)
            .then(function (response) { console.log(response); })
            .catch(function (error) {
                console.log(error);
            });
    }
}